Softeq.XToolkit.ProjectInfrastructure/HttpRequest.cs

```diff
@@ -49,6 +53,43 @@ public HttpRequest(Uri uri, HttpMethods method) : this(uri)
             CustomHeaders = new Dictionary<string, object>(headers ?? new Dictionary<string, object>());
         }
 
+        public object Data
+        {
+            get => _data;
+            private set
+            {
+                _serializeDataLock.Wait();
+
+                _data = value;
+                _serializedData = null;
+
+                _serializeDataLock.Release();
+            }
+        }
+
+        public string ContentType { get; private set; }
+
+        public async Task<string> GetSerializedDataAsync()
```

Comment:
It looks very strange.

Because:

1. Basic definition - class `HttpRequest` must be simple
2. Serialization logic must be separate
3. Thread-safe wrapper for serialization must be separate